### PR TITLE
Bugfix 215

### PR DESCRIPTION
**Pull request by carlospzurita**
Added parameter etf.showusername on properties file

Modified EtfConfigController to track new parameter

Modified TestRunController to pass etf.showusername as argument to each TestTaskDto on the TestRun

Fixed typo detected on ExecutableTestSuiteView.js for Start button

Cross-reference to https://github.com/etf-validator/etf-suitd/pull/2

### CODE REVIEW

**Comment by jonherrmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=5756548&bid=18843658)

File: src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java [hunk 1]
```diff
@@ -155,6 +157,8 @@
                     put(ETF_PARALLEL_EXECUTIONS, "" + Runtime.getRuntime().availableProcessors());
                     put(ETF_QUEUE_SIZE, "" + Runtime.getRuntime().availableProcessors() * 3);
 
+                    put(ETF_SHOWUSERNAME, "false");
```
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java:160]: Default should be true. Only your instance needs this.

File: src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java [hunk 2]
```diff
@@ -506,6 +507,11 @@ public void start(@RequestBody @Valid StartTestRunRequest testRunRequest, Bindin
                 }
             }
 
+            // Add etf.showusername property
+            for (TestTaskDto testTaskDto : testRunDto.getTestTasks()) {
+                testTaskDto.getArguments().setValue("etf.showusername", etfConfig.getProperty("etf.showusername"));
```
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java:512]: Why is this required?

File: src/main/webapp/js/app/views/ExecutableTestSuiteView.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ define([
                     markDependants(e.currentTarget.querySelectorAll("option")[1].value);
                 });
 
-                $("#fadin-start-tests-button").on("clic", function (e) {
+                $("#fadin-start-tests-button").on("click", function (e) {
```
Comment (jonherrmann) [inline src/main/webapp/js/app/views/ExecutableTestSuiteView.js:126]: 👍

File: src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java [hunk 4]
```diff
@@ -106,6 +106,8 @@
     private static final String ETF_PARALLEL_EXECUTIONS = "etf.testruns.threads.max";
     private static final String ETF_QUEUE_SIZE = "etf.testruns.queued.max";
 
+    public static final String ETF_SHOWUSERNAME = "etf.showusername";
```
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java:109]: I suggest ETF_SHOW_USERNAME = etf.show.username

File: src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java (hunk 1 above)
Comment (carlospzurita) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java]: Default value modified on latest commit

**Comment by jonherrmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/interactive-instruments/etf-webapp/pullRequest?prid=5756548&bid=18843658)

File: src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java (hunk 4 above)
Comment (carlospzurita) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java:109]: Key modified on latest commit

File: src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java (hunk 2 above)
Comment (carlospzurita) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java:512]: Our fix was relying on including the parameter on the TestTask to check it at the logging (as you can see on the pull request https://github.com/etf-validator/etf-suitd/pull/2) . To get this parameter from the ETFConfigController and pass it to the TestTask, we iterated over all the TestTasks in the TestRun and added the configuration parameter as an argument for the task.

Comment (jonherrmann) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java:512]: We have configuration properties and use them everywhere in ETF -also in the test drivers- to allow read access of these fixed values. Other developers would most likely never expect this configuration property in the Test Tasks. This change is hard to maintain and I see no reason why this code is necessary.

Comment (carlospzurita) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java:512]: Would it be ok if we move the change from the TestRunController to the SuiTestDriver, into the method createTestTask ?

Comment (jonherrmann) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java:512]: Since we wont be using SoapUI for the foreseeable future, code quality is not so important to me in that module. You are welcome to make the change there, although I woud prefer a change similar to the [other](https://github.com/interactive-instruments/etf-bsxtd/blob/next/src/main/java/de/interactive_instruments/etf/testdriver/bsx/BasexTestTask.java#L102) tds.

Comment (carlospzurita) [reply src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java]: We have updated the PR https://github.com/etf-validator/etf-suitd/pull/2 to reflect this change, and removed the functionality from the TestRunController

File: src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java (hunk 4 above)
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/EtfConfigController.java:109]: Please also adapt the variable name: `ETF_SHOWUSERNAME` -> `ETF_SHOW_USERNAME`

File: src/main/resources/etf-config.properties [hunk 5]
```diff
@@ -143,3 +143,7 @@ etf.webapp.base.url = http://localhost:8080/etf-webapp
 # Directory that the application uses to backup files
 # Default: bak
 # etf.backup.dir = bak
+
+# Prints user credentials (name and password) in log file
```
Comment (jonherrmann) [inline src/main/resources/etf-config.properties:147]: | Prints user credentials (name and password)

Thats not true. Regardless of your change, the password is not displayed (and it never was).

| in log file

Since you have security in mind with this change: this is too imprecise, an administrator might think that it is output in the etf.log file and enable this. You should write that this is the test run log that every user can access.
